python/tk_hiero_export/sg_shot_processor.py

```diff
@@ -256,6 +194,73 @@ def _get_tags(self, items):
         tags = [tag for tag in tags if "Nuke Project File" not in tag.name()]
         return tags
 
+class ShotgunShotProcessor(ShotgunHieroObjectBase, FnShotProcessor.ShotProcessor):
+    """
+    Adds hook functionality to the built in Shot processor.
+    """
+    def __init__(self, preset, submission=None, synchronous=False):
+        FnShotProcessor.ShotProcessor.__init__(self, preset, submission, synchronous)
+
+        # Call pre processor hook here to make sure it happens pior to any 'hook_resolve_custom_strings'.
+        # The order if execution is basically [init processor, resolve user entries, startProcessing].
+        self.app.execute_hook("hook_pre_export", processor=self)
```

Comment:
That's a good question. I see that it is passed to the pre_export hook, which I hadn't noticed before. That hook as we provide it is empty, so I'm not sure what prompted it to be provided. It's possible that we would break a hook someone has written themselves, but I'm not sure how much we can do about that. We could also pass in the new ProcessorUI object, but that would still potentially break hooks, though also provide the necessary data to for them to fix things themselves.

Thoughts? We can chat about it if you like. I'm open to suggestions!